File: src/Offering/Offering/Models/Product.cs
namespace Offering.Models;

public class Product
{
    public Guid Id { get; private set; }
    
    public string Name { get; private set; }
    
    public string Description { get; private set; }
    
    protected Product(Guid id, string name, string description)
    {
        Id = id;
        Name = name;
        Description = description;
    }
    
    public static Product Create(Guid id, string name, string description)
    {
        return new Product(id, name, description);
    }
}

File: src/OfferSB/src/main/java/com/example/offersb/models/Product.java
package com.example.offersb.models;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@NoArgsConstructor
public class Product
{
    @Getter
    @Id
    private UUID id;

    @Getter
    private String name;

    @Getter
    private String description;

    protected Product(UUID id, String name, String description)
    {
        this.id = id;
        this.name = name;
        this.description = description;
    }

    public static Product create(UUID id, String name, String description)
    {
        return new Product(id, name, description);
    }
}
